AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -105,27 +102,39 @@ - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
   return child;
 }
 
+#pragma mark - Parent
+
 - (void)setParent:(id<ASLayoutable>)parent
 {
-  // FIXME: Locking should be evaluated here.  _parent is not widely used yet, though.
-  _parent = parent;
+  {
+    ASDN::MutexLocker l(__instanceLock__);
+    _parent = parent;
+  }
   
   if ([parent supportsUpwardPropagation]) {
     ASEnvironmentStatePropagateUp(parent, self.environmentState.layoutOptionsState);
   }
 }
 
+- (id<ASLayoutable>)parent
+{
+  ASDN::MutexLocker l(__instanceLock__);
```

Comment:
@maicki I'm pretty sure that I'd removed locking from these methods due to deadlocks before (since they were rarely used).  We should be very cautious about adding them, maybe we can do that in a separate diff?

Super excited to land this soon, tonight would be great!